yew/src/virtual_dom/vtag.rs

```diff
@@ -33,22 +32,142 @@ pub const SVG_NAMESPACE: &str = "http://www.w3.org/2000/svg";
 /// Default namespace for html elements
 pub const HTML_NAMESPACE: &str = "http://www.w3.org/1999/xhtml";
 
-/// Used to improve performance of runtime element checks
-#[derive(Clone, Copy, Debug, PartialEq)]
-enum ElementType {
-    Input,
-    Textarea,
-    Button,
-    Other,
+/// Contains field only applicable to different Element types.
+/// Reduces the memory footprint by excluding impossible field combinations.
+/// Grouped partially in accordance to the
+/// [HTML spec](https://html.spec.whatwg.org/multipage/syntax.html#elements-2).
+#[derive(Debug, Clone)]
+pub enum VTagInner {
+    /// [InputElement](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
+    /// Is a [Void Element](https://html.spec.whatwg.org/multipage/syntax.html#void-elements).
+    Input {
+        /// Contains value property
+        value: Option<String>,
+        /// Represents `checked` attribute of
+        /// [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-checked).
+        /// It exists to override standard behavior of `checked` attribute, because
+        /// in original HTML it sets `defaultChecked` value of `InputElement`, but for reactive
+        /// frameworks it's more useful to control `checked` value of an `InputElement`.
+        checked: bool,
+    },
+
+    /// [TextAreaElement](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea)
+    TextArea {
+        /// Contains value property
+        ///
+        /// A
+        /// [TextAreaElement](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea)
+        /// can only contain a single text node so this doubles as the child list source.
```

Comment:
I may be wrong here. Just a note that the grammar feels a bit off.
```suggestion
        /// A textarea can only contain a single text node so this doubles as the child list source.
```